### ISSUE TITLE
Instance labels don't apply on disk

### ISSUE THREAD

**Issue by mouglou** [hashicorp/terraform-provider-google#8861]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to the `modular-magician` user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to `hashibot`, a community member has claimed the issue already.

### Terraform Version

$ terraform --version
Terraform v0.14.2

### Affected Resource(s)

* google_compute_instance

### Terraform Configuration Files

```tf
resource "google_compute_instance" "master" {
  name = var.hostname
  hostname = ${var.hostname}.${var.domain}
  machine_type = var.machine_type
  zone = random_shuffle.gce_zones.result

  metadata = { 
    startup-script        = module.startup-scripts.content
    startup-script-custom = file("files/gce_init.sh")
  }

  labels = merge(
    {   
      deployer = "terraform"
    },  
    var.labels
  )

  boot_disk {
    auto_delete = true
    initialize_params {
      image = var.node_image
      size  = var.node_disk_size
      type  = var.node_disk_type
    }   
  }

  network_interface {
    network            = data.google_compute_network.vpc_network.self_link
    subnetwork         = data.google_compute_subnetwork.vpc_subnetwork.self_link
    subnetwork_project = var.gcp_host_project
  }
}
```

### Expected Behavior

When an instance is created, the disk should have the same labels than the instance. 

### Actual Behavior

The instance is created, with his own disk. But the disk doesn't get the labels which have been set on the instance.

### Steps to Reproduce

1.Define a resource like the example
2.`terraform apply`
3. In GCP, go the Compute section, select Disks, you will see that the disk have no labels.

### Important Factoids

Why is ask for this ? When I check the billing section, the sort by labels is giving me good and bad value for compute service.
As the disk have no labels, its in a "Charge for user usage" section. 

### References

* https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_instance

**Comment by edwardmedia** [comment]
@mouglou from Terraform perspective, instances and disks are different resources. In your config, you chose the default disk(s) for your instance. And the labels are for instances. If you want to have labels on disks, can you see if [google_compute_disk](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_disk) works for you?

**Comment by mouglou** [comment]
hello @edwardmedia ! thanks for your feedback !
Yes they are different resources, but they are implicit. You can't create an instance without a disk. So I think the default behaviour should be that the disk get the same labels as the instance. Unless, as you say, you create a disk via the google_compute_disk resource.

We try with the dedicated disk resources but its not easy to manage. For somes service, in the instance we set the auto_delete option to false because we want to keep the disk even if we destroy the instance to be able to access to the data.

But with terraform, if I decalre a module "MariaDB", and call the module, it will create the disk and the instance, and then, if you comment your or want to destroy, it will destroy the disk too...

**Comment by edwardmedia** [comment]
@mouglou yes I do see `labels` on disks is available in the api. I am updating the issue label to `enhancement` so it will be backlog and triaged. 

https://cloud.google.com/compute/docs/reference/rest/v1/instances

**Comment by mouglou** [comment]
Cool !!! Thanks @edwardmedia for you help ! I will keep an eye the futur version of the provider. Until that moment, as you saw, I will use the API to add the same labels as the instance on the disk.

**Comment by yuryu** [comment]
I think this issue is related to (and has conflicting ideas with) #9097. I think having the explicit ability to set labels for the boot disk gives more flexibility, and also is backward-compatible.

**Comment by kkm000** [comment]
If you want identical labels, It's an extra line of code to define labels in a locals block binding and assign them both to the instance and the disk.

**Comment by roaks3** [comment]
This was implemented with https://github.com/GoogleCloudPlatform/magic-modules/pull/2106, but missing docs. Then the datasource docs were updated with https://github.com/GoogleCloudPlatform/magic-modules/pull/6326.

The only remaining item here is to update the resource docs, which will be almost exactly the same as the datasource.

Note that some of the discussion/work here is on https://github.com/hashicorp/terraform-provider-google/issues/9097, which is being closed now as a duplicate.

**Comment by roaks3** [comment]
b/249551103

**Comment by Chupaka** [comment]
@roaks3 
> The only remaining item here is to update the resource docs, which will be almost exactly the same as the datasource.

https://github.com/GoogleCloudPlatform/magic-modules/pull/6558
